quiz/01-rate-limit/quiz01/core/tests.py

```diff
@@ -0,0 +1,47 @@
+from django.test import TestCase
+from django.test import client
+import time
+# Create your tests here.
+
+class APITestCase(TestCase):
+
+    def setUp(self):
+        self.c = client.Client()
+    
+    # Unover rate limiting get test 
+    # will be get 200 status
+    def test_get_not_over(self):
+        time.sleep(2)
```

Comment:
`time.sleep(2)` is used to clean up the previous state.
however, that means a test runner would take 2 more seconds to be able to complete this test.
are we able to make it act like a state machine? like

```
serverstub.resetRateLimitConstraint($ip or not)

for i in range(1,101):
            resp = self.c.get('/api/')
```

Reply:
Indeed, it would be better to switch to that behavior. However, it's unfortunate that [django-ratelimit](https://django-ratelimit.readthedocs.io/en/stable/) doesn't seem to provide that functionality directly. Therefore, in the new version, we have created that functionality ourselves, which can be viewed reset function in the '`core/view`' . It has also been added to the tests, improving our testing process. 
result: 
```
resp = self.c.post('/reset/',{'group':'get', 'key':'ip', 'rate':'100/s', 'method':'GET'}) 

for j in range(1,101):
    resp = self.c.get('/api/') #request api
```